### PR TITLE
chore: rewrite Text component styles and consistency to as const types

### PR DESCRIPTION
**Pull request by KelvinOm**
## Description
Rewrite Text component to CSS modules

#### PR fixes following issue(s)
Fixes # (issue number)
#27479 

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

- [x] Manual
- [ ] JUnit
- [x] Jest
- [x] Cypress

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### CODE REVIEW

File: app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@import "../../../styles/colors.module.css";
+@import "../../../sharedTypes/colors/colors.module.css";
```
Comment (jsartisan) [inline app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css:1]: css file in `sharedTypes` seems confusing 🤔

Comment (jsartisan) [reply app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css:1]: I feel it should be just `shared`

File: app/client/packages/design-system/theming/src/typography/src/types.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ import nunitoSans from "@capsizecss/metrics/nunitoSans12pt";
 import appleSystem from "@capsizecss/metrics/appleSystem";
 import BlinkMacSystemFont from "@capsizecss/metrics/blinkMacSystemFont";
 
-export const fontMetrics = {
+export const FontMetrics = {
```
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/theming/src/typography/src/types.ts:15]: You forgot to mention another fourth option — type. In my opinion, it's more used as a type, but we also use it as a generic variable to iterate over values of this type.

File: app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css (hunk 1 above)
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css:1]: Yep, make sense.

File: app/client/packages/design-system/theming/src/typography/src/types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/theming/src/typography/src/types.ts]: fixed

File: app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css (hunk 1 above)
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css]: fixed

**Review by jsartisan — APPROVED** [review]

**Pull request merged**